### PR TITLE
chore: More IntelliJ fixes

### PR DESCRIPTION
**Pull request by magicus**
These are manually selected from all issues notified by the "Artemis" profile.

### CODE REVIEW

File: common/src/main/java/com/wynntils/core/config/ConfigManager.java [hunk 1]
```diff
@@ -260,7 +261,7 @@ private List<ConfigHolder> getConfigOptions(Configurable parent) {
     }
 
     public List<ConfigHolder> getConfigHolders() {
-        return CONFIG_HOLDERS;
+        return Collections.unmodifiableList(CONFIG_HOLDERS);
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/core/config/ConfigManager.java]: We only of of this in the wrong place and we will allocate a lot memory, won't we? Why do we need this anyways?

File: common/src/main/java/com/wynntils/features/user/CommandAliasesFeature.java [hunk 2]
```diff
@@ -63,35 +63,5 @@ public void onCommandPacket(CommandsPacketEvent event) {
...
-            CommandAlias that = (CommandAlias) other;
-            return originalCommand.equals(that.originalCommand) && aliases.equals(that.aliases);
-        }
-
-        @Override
-        public int hashCode() {
-            return Objects.hash(originalCommand, aliases);
-        }
-    }
+    private record CommandAlias(String originalCommand, List<String> aliases) {}
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/CommandAliasesFeature.java]: Isn't this used in @Config? That means GSON won't work.

**Review by kristofbolyai — COMMENTED** [review]
I can't say I like the unmodifiable collection creation everywhere

**Comment by magicus** [comment]
> I can't say I like the unmodifiable collection creation everywhere

I can leave that out of this PR, so we can discuss it separately. I think it makes sense to improve the robustness of the code, but I am willing to discuss it.

File: common/src/main/java/com/wynntils/features/user/CommandAliasesFeature.java (hunk 2 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/CommandAliasesFeature.java]: Ah, I see. Annoying that GSON cannot handle records. :( I'll have to revert it then. Good catch!

File: common/src/main/java/com/wynntils/core/config/ConfigManager.java (hunk 1 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/core/config/ConfigManager.java]: The API from ConfigManager was not particularly streamlines for its actual purpose. I made this return a `Stream` instead which gets rid of the possibility to accidentally modify the list. I also fixed some other API issues that I ran into while doing it. Overall, it looks much smoother now.

**Comment by magicus** [comment]
I think returning `Stream` instead of `List` when we do not want the original list to be able to be modified might be a better design choice in many cases. I'll look into that as an alternative, but in a separate PR.

**Comment by kristofbolyai** [comment]
> I think returning `Stream` instead of `List` when we do not want the original list to be able to be modified might be a better design choice in many cases. I'll look into that as an alternative, but in a separate PR.

Sounds good. But definitely remove the unmodifiable stuff for now.

**Comment by magicus** [comment]
>  we will allocate a lot memory, won't we? 

I just realized I need to clear out a possible misconception. Calling `Collections.unmodifiableList` will not waste any memory. The new list is just a view, a wrapper backed by the underlying list, that has methods trying to modify the list throw unsupported operation. So there is no (or at least negligible) memory spent. What we get is a slight performance penalty (if this happens in hot spots, the JVM will likely optimize that away) since a get call needs to be delegated once.

But the upside you get from this is that you cannot by mistake modify an array that is still internal to another object.

If we really want to modify it, and not modify the list belonging to the object, then we need to make a copy. I have not discovered any instances that this is the case, though. But if it were, then we could definitely have a bug on our hands if we did not copy it.

**Comment by kristofbolyai** [comment]
> > we will allocate a lot memory, won't we?
> 
> I just realized I need to clear out a possible misconception. Calling `Collections.unmodifiableList` will not waste any memory. The new list is just a view, a wrapper backed by the underlying list, that has methods trying to modify the list throw unsupported operation. So there is no (or at least negligible) memory spent. What we get is a slight performance penalty (if this happens in hot spots, the JVM will likely optimize that away) since a get call needs to be delegated once.
> 
> But the upside you get from this is that you cannot by mistake modify an array that is still internal to another object.
> 
> If we really want to modify it, and not modify the list belonging to the object, then we need to make a copy. I have not discovered any instances that this is the case, though. But if it were, then we could definitely have a bug on our hands if we did not copy it.

Okay. Performance is once concern, making the code look worse is another. I just wouldn't want to see us needing to add unmodifiable wrappers in half our classes. I think we should use streams where logical, and leave the rest to unmodifiable? (Or just use Kotlin :P ?)

**Comment by magicus** [comment]
Ok, let's skip the unmodifiable part. I agree that it makes the code look worse. Are you happy with the rest of the PR?

**Comment by kristofbolyai** [comment]
> Ok, let's skip the unmodifiable part. I agree that it makes the code look worse. Are you happy with the rest of the PR?

Yes, I didn't find anything else

**Comment by magicus** [comment]
I take that as an approval :)

**Pull request merged**